lib/course/middleware/assets_middleware.rb

```diff
@@ -0,0 +1,56 @@
+module Course
+  module Middleware
+    class AssetsMiddleware
+      attr_reader :app
+
+      REQUEST_METHOD_GET = "GET".freeze
+      PUBLIC_URL = "/public".freeze
+
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
+        if get?(env) && public_url?(env)
+          status, headers, body = handle_public_request(env)
+        else
+          status, headers, body = app.call(env)
+        end
+
+        [status, headers, body]
```

Comment:
If we know, that `app.call` returns an array of 3 values, and `handle_public_request` returns an array of 3 values, the code can be shorten a bit. 
```ruby
def call(env)
  if get?(env) && public_url?(env)
    handle_public_request(env)
  else
    app.call(env)
  end
end
```

Reply:
fixed